understanding/20/link-purpose-in-context.html

```diff
@@ -1,403 +1,187 @@
+         to understand the page.</p>
+      <p>The second aspect of this Criterion, <q>link text together with its <a>programmatically
+         determined link context</a></q>, primarily benefits assistive technology users.
+         If it's not possible to provide link text that, in isolation, conveys the purpose of the link,
+         authors must make sure that there is at least additional <em>context</em> for the link.
+         Assistive technologies can provide users with easy access to the link context, provided that
+         it is programmatically determinable. This can give users additional information that will help
+         them determine the purpose of a link.</p>
       <p>The text of, or associated with, the link is intended to describe the purpose of the
          link. In cases where the link takes one to a document or a web application, the name
          of the document or web application would be sufficient to describe the purpose of
          the link (which is to take you to the document or web application). Note that it is
          not required to use the name of the document or web application; other things may
-         also describe the purpose of the link. 
-      </p>
-      
+         also describe the purpose of the link.</p>
       <p>
-         
-         <a href="page-titled" class="understanding">Success Criterion 2.4.2</a> deals with the titles of pages. Here also, the name of a document or web application
+         <a href="page-titled" class="understanding">Success Criterion 2.4.2</a> deals with the titles of pages. Here
+         also, the name of a document or web application
          being presented on the page would be sufficient to describe the purpose of the page.
          Having the link and the title agree, or be very similar, is good practice and provides
-         continuity between the link 'clicked on' and the web page that the user lands on.
-         
-      </p>
-      
+         continuity between the link 'clicked on' and the web page that the user lands on.</p>
       <p>In some situations, authors may want to provide part of the description of the link
          in logically related text that provides the context for the link. In this case the
          user should be able to identify the purpose of the link without moving focus from
          the link. In other words, they can arrive on a link and find out more about it without
          losing their place. This can be achieved by putting the description of the link in
-         the same sentence, paragraph, list item, or table cell as the link, or in the table header cell for a link in a data table, because these are directly associated with the link itself. Alternatively, authors may choose to use an ARIA technique to associate additional
-         text on the page with the link.
-      </p>
-      
+         the same sentence, paragraph, list item, or table cell as the link, or in the table header cell for a link in a
```

Comment:
I know the [H80](https://www.w3.org/WAI/WCAG22/Techniques/html/H2.html) (providing programmatic context via the preceding heading) has been demoted to advisory some years back, and I also know that I am not the only one to disagree with that decision. On the basis that Techniques are all informational content and the practice to provide context via the preceding heading seems both widespread and certainly no less obscure or less understood than things like using the table header to provide context, I wonder if there is a chance to re-promote H80. This would allow us to also mention headings here as a way to provide programmatic context. (I realize this would require a different PR and possibly drag on for quite some time, so I am all for going with these changes before we get there).

Reply:
Yeah suggest doing this as a separate PR, as this would likely stall this entire effort in endless discussions. (also noting that it's not that using table headers is also obscure, but rather that by default SRs provide a way to - without moving focus - have the relevant table headers that relate to the current cell announced, whereas for "what's the preceding heading" - while eminently doable from a structure point of view - is not an option you currently get, so not "accessibility supported" as such?)

Reply:
@patrickhlauke table headers / headings - yes, I see your point.

Reply:
FWIW - with JAWS - you can press JAWSKey+T to hear the preceding heading (in additional to page title) from where you are on the page.  It also tells you the heading level.

Reply:
in which case @mraccess77 we can probably reopen the discussion (separately) about whether or not the technique to use headings as context is advisory or sufficient...

Reply:
https://github.com/w3c/wcag/issues/1097 is probably the last/latest separate issue about this where discussion can continue